components/sidebar/sidebar.jsx

```diff
@@ -0,0 +1,692 @@
+                this.updateScrollbarOnChannelChange(nextChannel);
+            }
+            this.isSwitchingChannel = false;
+        }
+    }
+
+    getDisplayedChannels = () => {
+        return this.getDisplayedChannelsForProps(this.props);
+    }
+
+    getDisplayedChannelsForProps = (props) => {
+        return props.favoriteChannelIds.concat(props.publicChannelIds).concat(props.privateChannelIds).concat(props.directAndGroupChannelIds);
+    }
+
+    channelIdIsDisplayedForProps = (props, id) => {
+        const allChannels = this.getDisplayedChannelsForProps(props);
+        for (let i = 0; i < allChannels.length; i++) {
+            if (allChannels[i] === id) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    showMoreChannelsModal = () => {
+        this.setState({showMoreChannelsModal: true});
+        trackEvent('ui', 'ui_channels_more_public');
+    }
+
+    hideMoreChannelsModal = () => {
+        this.setState({showMoreChannelsModal: false});
+    }
+
+    showNewChannelModal = (type) => {
+        this.setState({newChannelModalType: type});
+    }
+
+    hideNewChannelModal = () => {
+        this.setState({newChannelModalType: ''});
+    }
+
+    showMoreDirectChannelsModal = (startingUsers) => {
+        trackEvent('ui', 'ui_channels_more_direct');
+        this.setState({showDirectChannelsModal: true, startingUsers});
+    }
+
+    hideMoreDirectChannelsModal = () => {
+        this.setState({showDirectChannelsModal: false, startingUsers: null});
+    }
+
+    openQuickSwitcher = (e) => {
+        e.preventDefault();
+        AppDispatcher.handleViewAction({
+            type: ActionTypes.TOGGLE_QUICK_SWITCH_MODAL
+        });
+    }
+
+    createSidebarChannel = (channelId, index) => {
+        const channel = this.props.allChannels[channelId];
+
+        if (channelId !== this.props.currentChannel.id && this.props.unreadChannelIds.includes(channelId)) {
+            if (!this.firstUnreadChannel) {
+                this.firstUnreadChannel = channelId;
+            }
+            this.lastUnreadChannel = channelId;
+        }
+
+        if (this.props.memberships[channel.id] && this.props.unreads.mentions) {
+            this.badgesActive = true;
+        }
+
+        return (
+            <SidebarChannel
+                key={channel.id}
+                ref={channel.id}
+                channelId={channel.id}
+                index={index}
+                membership={this.props.memberships[channel.id]}
+                active={channel.id === this.props.currentChannel.id}
+                loadingDMChannel={this.state.loadingDMChannel === index && channel.type === Constants.DM_CHANNEL}
+                currentTeamName={this.props.currentTeam.name}
+                currentUserId={this.props.currentUser.id}
+            />
+        );
+    }
+
+    render() {
+        this.updateTitle();
```

Comment:
If we can pull this out and only do it when the props change that would be great. It's very un-React-like to have side effects like this in the render function and it might give us a perf boost if we're rendering more often than just on channel switch

It might be better off if componentDidUpdate() than componentWillReceiveProps(), not sure though

I know this PR is huge already so if we were doing this before then I'm OK with just splitting this (and maybe some other stuff) into improvement tickets

Reply:
I have no problem doing this. But I'm not sure if it a side-effect or "not-rendering" thing. It only modifies the document.title which is, in some way, part of the rendering process. We aren't changing the global state, the local state, or any object attribute, it is only reading the props and "drawing" the title based on it.

Move it to the componentDidUpdate is trivial, but I'm not sure is the correct solution.

Reply:
I finally moved it to the componentDidUpdate, but I'm not sure is the correct place, as I said, IMHO it is part of the rendering process, other thing is that don't want to "re-render" this part each time.

Other thing is that the location of the title management inside the sidebar is weird, because is something global, to the application, not extrictly dependand on the sidebar. I think can be a Redux component called from the root component. There are other places where is used, but aren't too much and I think we can infer the title from the route + the global state.